WalletWasabi.Documentation/Guides/DeterministicBuildGuide.md

```diff
@@ -1,67 +1,79 @@
 
-# 2. Reproduce Builds
+**Warning:** Reproducible builds were introduced in [1.1.3 release](https://github.com/zkSNACKs/WalletWasabi/releases/tag/v1.1.3), you cannot use these instructions for older versions!
+
+## 1. Assert correct environment
+
+In order to reproduce Wasabi's builds, you need [git](https://git-scm.com/) package, Windows 10, and the version of .NET Core SDK that was used by the Wasabi team to produce given Wasabi Wallet release. The latest version of .NET Core SDK is always used, unless specified otherwise in the release notes of Wasabi Wallet. You can get it here https://dotnet.microsoft.com/download/dotnet-core.
+
+## 2. Reproduce builds
+
+You can see the list of Wasabi releases here: https://github.com/zkSNACKs/WalletWasabi/releases. Please note that each release has a version and a git hash assigned. The git hash is useful in the following instructions:
 
 ```sh
 git clone https://github.com/zkSNACKs/WalletWasabi.git
-git checkout {hash of the release} # This works from 1.1.3 release, https://github.com/zkSNACKs/WalletWasabi/releases
-cd WalletWasabi/WalletWasabi.Packager/
+cd WalletWasabi
+git checkout <SHA-1-hash-of-the-release>
+cd WalletWasabi.Packager
 dotnet restore
 dotnet build
 dotnet run -- --onlybinaries
 ```
 
-This will build our binaries for Windows, OSX and Linux from source code and open them in a file explorer for you.
+These commands will produce Wasabi's binaries for Windows, macOS and Linux from source code. Also, for your convenience, a new file explorer window will navigate you to the binaries location - i.e. `WalletWasabi\\WalletWasabi.Gui\\bin\\dist`.
 
 ![](https://i.imgur.com/8XAQzz4.png)
 
-# 3. Verify Builds
+## 3. Verify builds
 
-You can compare our binaries with the downloads we have on the website: https://wasabiwallet.io/
-In order to end-to-end verify all the downloaded packages you need a Windows, a Linux, and an OSX machine.
+Now, we will attempt to verify the binaries you have just compiled with the officially distributed binaries on https://wasabiwallet.io website. Please download those packages from the website, you should see the following files in your File Explorer:
 
 ![](https://i.imgur.com/aI9Kx0c.png)
 
-## Windows
+Please note that to completely verify Wasabi packages for all supported platforms, you actually need machines with Windows, Linux and macOS operating systems. If you don't have so many physical machines, you can use any virtualization sofware packages like [VirtualBox](https://www.virtualbox.org/), [VMWare](https://www.vmware.com/), etc.
 
-After you installed Wasabi from the `.msi`, it will be in `C:\Program Files\WasabiWallet` folder. You can compare it with your build:
+### Windows
 
-```sh
-git diff --no-index win7-x64 "C:\Program Files\WasabiWallet"
-```
+* Install Wasabi using `Wasabi-<version>.msi` file. It will install to `C:\Program Files\WasabiWallet` directory.
```

Comment:
wtf! This is crazy... someone doing this wants to be sure he will not being installing something that is not in the code (or malware). We cannot say install it first verify it later. 

msi allows you to extract the content. The user should extract the files and compare.

Reply:
* I have tried `7-zip` on my Windows machine and I cannot extract it.
* Also, I have tried https://thebackroomtech.com/2007/08/23/howto-extract-files-from-a-msi-file-using-the-windows-command-line/ (`msiexec /a PathToMSIFile /qb TARGETDIR=DirectoryToExtractTo`)

So your point is valid but I kind of think that this somehow is linked to how the msi is packaged via Wasabi Packager. 

Can anyone confirm that 7-zip or that msiexec tool are unable to extract `Wasabi.msi` file?

Reply:
I don't have a windows but I have extracted msi files in the past many times so, i am 100% sure it can be done. Of course that I don't know if that can be different for our .msi or if extracting only works in some situations or for some specific cases.

Reply:
![image](https://user-images.githubusercontent.com/58662979/88224477-518c4100-cc69-11ea-945b-7c95387cdab1.png)

I can see this on my machine.

Reply:
There are various ways to extract `.msi` files:

## Simple 7-zip

Does not work on Wasabi .msi file.

## `Dark.exe` from Wix toolset

See https://fredrikwall.se/deployment/how-to-extract-a-wix-toolset-package/

```
.\wix311-binaries\dark.exe .\Wasabi-1.1.11.1.msi -x WasabiXXX
```

The result is similar to the picture above. 

## `lessmsi` (free, open-source)

> A tool to view and extract the contents of a Windows Installer (.msi) file.

git repo: https://github.com/activescott/lessmsi
project: http://lessmsi.activescott.com/

```
.\lessmsi-v1.6.91\lessmsi.exe x Wasabi-1.1.11.1.msi "WasabiXXX\"
```

This actually works for me.

btw: https://github.com/zkSNACKs/WalletWasabi/blob/d08c8485f14104379c287ccca2893749f8b8e862/WalletWasabi.WindowsInstaller/Product.wxs#L16 - Can this possibly affect the final `.msi` file?

Reply:
@kiminuo I can extract using 7-Zip.

Reply:
@yahiheb What version of 7-zip you have?